core/trino-spi/src/main/java/io/trino/spi/metrics/Metrics.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.spi.metrics;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+import static java.util.Objects.requireNonNull;
+
+public class Metrics
+{
+    public static final Metrics EMPTY = new Metrics(Map.of());
+
+    private final Map<String, Metric> metrics;
+
+    @JsonCreator
+    public Metrics(Map<String, Metric> metrics)
+    {
+        this.metrics = Map.copyOf(requireNonNull(metrics, "metrics is null"));
+    }
+
+    public static Metrics merge(Metrics... metrics)
```

Comment:
Since we only give two metrics as input, it might be better to simplify this method to just take 2 metrics as input.
Or you could even make this class implement `Mergeable<Metrics>` interface.

Reply:
Made `Metrics` mergeable.